### PR TITLE
Added `SubscriptionPeriod` to `APITester`s

### PR DESCRIPTION
**Pull request by NachoSoto**
Also added missing fields for `@objc`.

### CODE REVIEW

File: APITesters/SwiftAPITester/SwiftAPITester/SubscriptionPeriodAPI.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func checkSubscriptionPeriodUnit() {
+    let unit: SubscriptionPeriod.Unit = .unknown
+
+    switch unit {
+    case
+            .unknown,
+            .day,
+            .week,
+            .month,
+            .year:
```
Comment (taquitos) [inline APITesters/SwiftAPITester/SwiftAPITester/SubscriptionPeriodAPI.swift:30]: formatting looks weird here, but it's what xcode does 🤷‍♂️

File: Purchases/Purchasing/StoreKitAbstractions/SubscriptionPeriod.swift [hunk 2]
```diff
@@ -17,8 +17,8 @@ import StoreKit
 @objc(RCSubscriptionPeriod)
 public class SubscriptionPeriod: NSObject {
 
-    public let value: Int
-    public let unit: Unit
+    @objc public let value: Int
```
Comment (taquitos) [inline Purchases/Purchasing/StoreKitAbstractions/SubscriptionPeriod.swift:20]: 😍

**Review by taquitos — APPROVED** [review]
👯 🐐🐐

File: Purchases/Purchasing/StoreKitAbstractions/SubscriptionPeriod.swift (hunk 2 above)
Comment (aboedo) [reply Purchases/Purchasing/StoreKitAbstractions/SubscriptionPeriod.swift:20]: great catches here

**Review by aboedo — APPROVED** [review]

File: APITesters/SwiftAPITester/SwiftAPITester/SubscriptionPeriodAPI.swift (hunk 1 above)
Comment (NachoSoto) [reply APITesters/SwiftAPITester/SwiftAPITester/SubscriptionPeriodAPI.swift:30]: Yeah if I put `.unknown` in the `case` line they don't align 🤦‍♂️

**Pull request merged**